tests/gui/gui_specific/test_gui.py

```diff
@@ -109,3 +110,39 @@ def on_an_action(state, id, payload):
     with gui.get_flask_app().app_context():
         gui._Gui__on_action(an_id, a_non_action_payload) # type: ignore[attr-defined]
         gui._Gui__on_action(an_id, an_action_payload) # type: ignore[attr-defined]
+
+
+def test__notification_on_close(gui: Gui):
+    notification_id = "test-id"
+    
+    def on_notification_closed(state, notification_id, reason=None):
+        assert notification_id == "test-id"
+        assert reason == "timeout"
+    
+    gui._set_frame(inspect.currentframe())
+
+    
+    gui._Gui__send_ws_notification = MagicMock()
```

Comment:
there's a way to get the ws message sent without mocking a private method
Check https://github.com/Avaiga/taipy/blob/111f0601bc819b1f0aa2bf71b92ef7455e28b114/tests/gui/actions/test_navigate.py#L28